bam_addrprg.c

```diff
@@ -343,11 +358,13 @@ static bool init(const parsed_opts_t* opts, state_t** state_out) {
 
     retval->output_header = sam_hdr_dup(retval->input_header);
 
-    if (opts->uncompressed)
+    if (opts->uncompressed) {
         strcat(output_mode, "0");
-    if (opts->output_name) // File format auto-detection
+    }
+    if (opts->output_name) { // File format auto-detection
         sam_open_mode(output_mode + strlen(output_mode),
                       opts->output_name, NULL);
+    }
```

Comment:
You mention "ensure non-trivial ifs are explicitly {}'d".  However both of these are single statements rather than compound ones.

Do you define a trivial if as one on the same line, rather than this?

```
if (foo)
    bar();
```

Generally our coding style is single-statements don't require bracing.  I don't object if people do it for new code as it's a minor thing, but I don't see the benefit of changing old code to be that like, especially when there are other single statement if's left unchanged, eg your own `if (!cleanup_state)` below.

Reply:
to be fair the change to `if (opts->uncompressed)` I'm happy to drop. It was the multiline nature of sam_open_mode that gave me the urge to {} it